### PR TITLE
Bug 1368632 - Added DevEdition to the repositories to parse in ftpscr…

### PR DESCRIPTION
**Pull request by adngdb**
…aper.

This new repository contains information about the builds of the freshly created DevEdition channel. We need to parse it in order to be able to show correct version numbers in Socorro.

Asking @peterbe for a review because you're the one who has the most knowledge about the ftpscraper.

### CODE REVIEW

**Comment by willkg** [comment]
@mostlygeek Do you know if buildhub covers this directory and devedition builds?

**Comment by mostlygeek** [comment]
I don't think so. I opened up a BuildHub issue (https://github.com/mozilla-services/buildhub/issues/162) to back fill the data and continue scraping it going forward. 

I wonder if devedition build id's == beta build id's now?

File: socorro/cron/jobs/ftpscraper.py [hunk 1]
```diff
@@ -333,7 +333,7 @@ def scrape_json_releases(self, connection, product_name):
         for directory in ('nightly', 'candidates'):
             try:
                 url, = self.get_links(prod_url, starts_with=directory)
-            except IndexError:
+            except (IndexError, ValueError):
```
Comment (willkg) [inline socorro/cron/jobs/ftpscraper.py]: What kicks up a `ValueError`?

Comment (peterbe) [inline socorro/cron/jobs/ftpscraper.py]: Makes you wonder, what *kickED* the `IndexError` before? Terrible code :)

**Comment by peterbe** [comment]
Hard to review, other than, Yes! it works. 
Here are the builds it finds:

```
▶ python socorro/cron/jobs/ftpscraper.py --product=devedition
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - app_name: SocorroAppBaseClass
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - app_version: 1.0
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - current configuration:
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - application: <class 'socorro.cron.jobs.ftpscraper.FTPScraperCronAppDryRunner'>
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - base_url: https://archive.mozilla.org/pub/
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - connect_timeout: 3.5
2017-06-16 08:20:47,304 INFO - SocorroAppBaseClass -  - MainThread - crontabber_job_class: <class 'socorro.cron.jobs.ftpscraper.FTPScraperCronApp'>
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - database.database_class: <MagicMock id='4548326480'>
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - database.database_transaction_executor_class: <class 'crontabber.transaction_executor.TransactionExecutor'>
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - date: 2017-06-16
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - dry_run: False
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - json_files_to_ignore: ['*.mozinfo.json', '*test_packages.json']
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logger: <socorro.app.socorro_app.LoggerWrapper object at 0x10f21d090>
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.stderr_error_logging_level: 10
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.syslog_error_logging_level: 40
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.syslog_facility_string: user
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.syslog_host: localhost
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - logging.syslog_port: 514
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - products: ('devedition',)
2017-06-16 08:20:47,305 INFO - SocorroAppBaseClass -  - MainThread - read_timeout: 10
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.stderr_error_logging_level: 10
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.stderr_line_format_string: {asctime} {levelname} - {app_name} - {message}
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.syslog_error_logging_level: 40
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.syslog_facility_string: user
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.syslog_host: localhost
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.syslog_line_format_string: {app_name} (pid {process}): {asctime} {levelname} - {threadName} - {message}
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.logging.syslog_port: 514
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.postgresql.database_class: <class 'crontabber.connection_factory.ConnectionFactory'>
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - resource.postgresql.database_transaction_executor_class: <class 'crontabber.transaction_executor.TransactionExecutor'>
2017-06-16 08:20:47,306 INFO - SocorroAppBaseClass -  - MainThread - retries: 5
2017-06-16 08:20:47,306 DEBUG - SocorroAppBaseClass -  - MainThread - scraping devedition releases for date 2017-06-16
2017-06-16 08:20:47,988 DEBUG - SocorroAppBaseClass -  - MainThread - Dir nightly not found for devedition
INSERT BUILD
('devedition', '54.0', 'linux', u'20170525225011', 'beta', '11', u'mozilla-beta', 'build4')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170525225011', 'beta', '11', u'mozilla-beta', 'build4')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'mac', u'20170525155044', 'beta', '11', u'mozilla-beta', 'build4')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170525155044', 'beta', '11', u'mozilla-beta', 'build4')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170525155044', 'beta', '11', u'mozilla-beta', 'build4')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170529095024', 'beta', '12', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170529095024', 'beta', '12', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'mac', u'20170529025115', 'beta', '12', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170529025115', 'beta', '12', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170529025115', 'beta', '12', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170601203257', 'beta', '13', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170601203257', 'beta', '13', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'mac', u'20170601133324', 'beta', '13', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170601133324', 'beta', '13', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170601133324', 'beta', '13', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170605195640', 'beta', '14', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'linux', u'20170605195640', 'beta', '14', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'mac', u'20170605125726', 'beta', '14', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170605125726', 'beta', '14', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '54.0', 'win', u'20170605125726', 'beta', '14', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'linux', u'20170613054006', 'beta', '1', u'mozilla-beta', 'build5')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'linux', u'20170613054006', 'beta', '1', u'mozilla-beta', 'build5')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'mac', u'20170612224034', 'beta', '1', u'mozilla-beta', 'build5')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'win', u'20170612224034', 'beta', '1', u'mozilla-beta', 'build5')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'win', u'20170612224034', 'beta', '1', u'mozilla-beta', 'build5')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'linux', u'20170615140023', 'beta', '2', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'linux', u'20170615140023', 'beta', '2', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'mac', u'20170615070049', 'beta', '2', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'win', u'20170615070049', 'beta', '2', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
INSERT BUILD
('devedition', '55.0', 'win', u'20170615070049', 'beta', '2', u'mozilla-beta', 'build1')
{'ignore_duplicates': True}
```

**Comment by willkg** [comment]
Regarding the `ValueError`, I was just curious. I don't think that should block landing this. ftpscraper isn't going to be around much longer.

File: socorro/cron/jobs/ftpscraper.py (hunk 1 above)
Comment (adngdb) [inline socorro/cron/jobs/ftpscraper.py]: If the result of ``self.get_links`` is empty, the ``url, =`` thing cannot work and raises the ``ValueError``. As Peter said, I wonder what triggered the ``IndexError``, but I honestly didn't want to spend any time looking into that... :)

**Pull request merged**